### ISSUE TITLE
Invalid internet address

### ISSUE THREAD

**Issue by K1yoshiSho** [flutter/flutter#126650]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

If send request i have problem with address. But before the upgrade to latest version (3.10) i dosnt have this issue.

### Expected results

Valid address.

### Actual results

Error Invalid internet address

### Code sample

main() async {
  var client = HttpClient();
  client.openUrl("get", Uri.parse("http://ptsv2.com/t/e2xwh-1622643201"));

  return runApp(MaterialApp(home: Scaffold(
    body: Center(child: Text("Hello")),
  )));
}

### Flutter Doctor output

[√] Flutter (Channel stable, 3.10.0, on Microsoft Windows [Version 10.0.22621.1555], locale ru-KZ)
    • Flutter version 3.10.0 on channel stable at C:\Flutter\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 84a1e904f4 (3 days ago), 2023-05-09 07:41:44 -0700
    • Engine revision d44b5a94c9
    • Dart version 3.0.0
    • DevTools version 2.23.1

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at C:\Users\yelam\AppData\Local\Android\sdk
    • Platform android-33-ext4, build-tools 33.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-9505619)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Professional 2022 17.5.4)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Professional
    • Visual Studio Professional 2022 version 17.5.33530.505
    • Windows 10 SDK version 10.0.22000.0

[√] Android Studio (version 2022.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-9505619)

[√] VS Code (version 1.78.1)
    • VS Code at C:\Users\yelam\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.64.0

[√] Connected device (4 available)
    • sdk gphone64 x86 64 (mobile) • emulator-5554 • android-x64    • Android 13 (API 33) (emulator)
    • Windows (desktop)            • windows       • windows-x64    • Microsoft Windows [Version 10.0.22621.1555]
    • Chrome (web)                 • chrome        • web-javascript • Google Chrome 112.0.5615.138
    • Edge (web)                   • edge          • web-javascript • Microsoft Edge 113.0.1774.35

[√] Network resources
    • All expected network resources are available.

• No issues found!

**Comment by K1yoshiSho** [comment]
This error is mentioned here: https://github.com/flutter/flutter/issues/83792

But this error has returned again and flutter upgrade does not help to fix this error.

**Comment by darshankawar** [comment]
@K1yoshiSho Thanks for the report. 
I ran the code sample on Android emulator using latest stable and master versions, but didn't get the reported error.

<img width="220" alt="Screenshot 2023-05-12 at 4 04 44 PM" src="https://github.com/flutter/flutter/assets/67046386/aefd2be5-503b-4fb5-a96c-95d8142ee8e4">

If you have more details to properly replicate, please provide it here. Also provide the complete error log.

**Comment by K1yoshiSho** [comment]
Can you check please this URL:
https://qt.workspace.kz

**Comment by K1yoshiSho** [comment]
> @K1yoshiSho Thanks for the report. 
> I ran the code sample on Android emulator using latest stable and master versions, but didn't get the reported error.
> 
> <img width="220" alt="Screenshot 2023-05-12 at 4 04 44 PM" src="https://github.com/flutter/flutter/assets/67046386/aefd2be5-503b-4fb5-a96c-95d8142ee8e4">
> 
> If you have more details to properly replicate, please provide it here. Also provide the complete error log.
> 

Can you check please this URL:
https://qt.workspace.kz

**Comment by darshankawar** [comment]
I still see the same result as I posted earlier using the new link you shared above.

**Comment by K1yoshiSho** [comment]
> I still see the same result as I posted earlier using the new link you shared above.

@darshankawar 
![image](https://github.com/flutter/flutter/assets/96978512/6a0c13f8-7ecb-4603-92f2-3a273a52db5f)

address is qt.workspace.kz

inAddr is null. I send REST api request to this URL and every request is with error.

**Comment by K1yoshiSho** [comment]
@darshankawar 
But request without error
![image](https://github.com/flutter/flutter/assets/96978512/c4e05bde-bc6b-48c1-8dce-affb84027f13)

**Comment by Mayb3Nots** [comment]
> @K1yoshiSho Thanks for the report. I ran the code sample on Android emulator using latest stable and master versions, but didn't get the reported error.
> 
> <img alt="Screenshot 2023-05-12 at 4 04 44 PM" width="220" src="https://user-images.githubusercontent.com/67046386/237948113-aefd2be5-503b-4fb5-a96c-95d8142ee8e4.png">
> If you have more details to properly replicate, please provide it here. Also provide the complete error log.

Hi @darshankawar did you try enabling all exceptions on vscode? The network request does successfully complete however an exception is thrown and caught in the framework it self. It also seems to happen on evey network call.

**Comment by milindgoel15** [comment]
The issue only happens to be occurring after upgrading to 3.10. The same problem happens on HTTP calls to open weather API.

![image](https://github.com/flutter/flutter/assets/45682747/b55a09a2-2e91-4ffb-8700-6c87733d7861)

**Comment by justinekatebe69** [comment]
I agree with @milindgoel15 after upgrading to flutter 3.10.0, I've experiencing the same issue, but when I turn off exception detection, it works just fine. 

<img width="218" alt="Screenshot 2023-05-12 at 20 51 08" src="https://github.com/flutter/flutter/assets/67024464/e9ddf1ba-5a90-4d6a-bf9b-ccf9c7b3c941">

**Comment by justinekatebe69** [comment]
I agree with @milindgoel15 after upgrading to flutter 3.10.0, I've experiencing the same issue, but when I turn off exception detection, it works just fine. 

<img width="218" alt="Screenshot 2023-05-12 at 20 51 08" src="https://github.com/flutter/flutter/assets/67024464/e9ddf1ba-5a90-4d6a-bf9b-ccf9c7b3c941">

**Comment by darshankawar** [comment]
Thanks for the update and feedback. I don't use VSCode, but AS. Since it seems to be happening on latest stable version per reports, is there a way for anyone to verify if the same persists on latest master version as well ?

**Comment by milindgoel15** [comment]
yes, the same issue exists on the master channel. tested through vs code.
![image](https://github.com/flutter/flutter/assets/45682747/a849b817-1088-4e5e-8beb-dfa51f382709)

![image](https://github.com/flutter/flutter/assets/45682747/685f3ae3-72d2-46fa-a00f-dd75d498ec85)

**Comment by darshankawar** [comment]
Thanks for the feedback and the reports. I am unable to replicate this as this seems to be more prominent using VSCode.
I am going ahead and labeling this for team's attention and a potential regression based on the reports as it seems it started to occur after upgrade to 3.10.

**Comment by K1yoshiSho** [comment]
> /cc @natebosch for thoughts if this is due to http package and @DanTup for vscode related input.
> 
> 

This issue had in http package and dio package too

**Comment by K1yoshiSho** [comment]
> I can reproduce this in VS Code if I tick the "All Exceptions" tickbox on the debug side bar, but as far as I can tell this is working as expected.
> 
> The "All Exceptions" tickbox means "pause the debugger whenever an exception is thrown, _even if it will be caught by a `catch` block". This does not necessarily indicate an error, it is a tool for debugging (for example if you think an exception is occurring but is being caught somewhere, this will allow you to step through this).
> 
> For example, given the code:
> 
> ```dart
> try {
>   throw 'handled error';
> } catch (e) {
>  // ignore errors
> }
> ```
> 
> If you tick "All Exceptions", the debugger will pause on the `throw` statement. If you only tick "Uncaught Exceptions", it will not. For general debugging, you should only tick "Uncaught Exceptions". Use "All Exceptions" when you are specifically trying to debug exceptions you think are occurring but something is (perhaps incorrectly) catching.
> 
> In this particular case, the exception is caught on the next stack frame up:
> 
> <img width="992" alt="Screenshot 2023-05-15 at 14 39 18" src="https://github.com/flutter/flutter/assets/1078012/952c6292-d1b3-4350-b97d-21418765285e">
> 
> So the exception is being handled and is nothing to worry about.
> 
> As for why it's only started happening recently, my guess is that it's because of some recent changes that just happen to cause there to be a handled exception in this code that wasn't there before. Probably this is from https://github.com/dart-lang/sdk/commit/d884d1a76e47e45c5b34b376575b10e236ebeb0a which contains a change which introduces a call to `_InternetAddress.tryParse` which may throw (and catch) an exception for non-IP address host strings:
> 
> ```
>       if (host is String) {
>         // Attempt to interpret the host as a numeric address
>         // (e.g. "127.0.0.1"). This will prevent [InternetAddress.lookup] from
>         // generating an unnecessary address in a different address family e.g.
>         // `InternetAddress.lookup('127.0.0.1', InternetAddressType.IPv6)`
>         // may return `InternetAddress('::ffff:127.0.0.1').
>         host = _InternetAddress.tryParse(host) ?? host;
>       }
> ```
> 
> If anyone is seeing this _without_ ticking "All Exceptions", please let me know. But if you have "All Exceptions" ticked, I believe this is all working as expected.
> 

I understand you, thank you! But can it be done the way it was before so that it doesn't react even if the setting is "All exceptions"?

**Comment by DanTup** [comment]
@K1yoshiSho 

> But can it be done the way it was before so that it doesn't react even if the setting is "All exceptions"?

In theory, it's probably possible to rewrite the code to not throw (and catch) an exception. I'm not familiar with that code or the recent changes though, so I couldn't say how simple (or worthwhile) it would be. I would generally recommend not using "All Exceptions" for general debugging though, and only enable it when you're specifically trying to track down something that may be a _caught_ exception.


@milindgoel15 that looks like a different exception. Can you provide a code sample to reproduce it?

**Comment by milindgoel15** [comment]
This is my open weather API code:

```dart
    final currentQueryParameters = {
      'q': 'london',
      'appid': API_KEY,
      'units': "metric,
      'lang': languageCode,
    };
    final currentUri = Uri.https(
      'api.openweathermap.org',
      '/data/2.5/weather',
      currentQueryParameters,
    );
    final currentRes = await http.get(currentUri);
    dynamic currentData;

    if (currentRes.statusCode == 200) {
      currentData = jsonDecode(currentRes.body);
    }
```
And i am using riverpod to create a future provider:
```dart
final weatherProvider = Provider((ref) => WeatherAPIResponse());

final fetchCurrentProvider = FutureProvider.family(
  (ref, String location) {
    final weatherResponse = ref.watch(weatherProvider);
    return weatherResponse.getWeather(location);
  },
);
```

**Comment by DanTup** [comment]
@milindgoel15 could you provide a complete code example I can run? It's not clear to me how to combine those snippets of code (I'm not familiar with Riverpod, but I don't see any exception just running the `http.get()` code above).


> If we enable the "all exceptions" in this case, the invalid internet address exception would be caused again, making it difficult to debug.

I agree that's not ideal - debugging caught exceptions is always going to be difficult, but it's not always feasible to eliminate all instances of caught exceptions. Like I said, I can't really comment on the code above, it's outside my area. You could file an issue at https://github.com/dart-lang/sdk about changing that specific code to not throw/catch exceptions in that way - it probably is an exception that will be triggered fairly often if it occurs for every `get` of ever hostname, so perhaps it would be worth trying to change.


> It's really tough to debug with this, because some of the SDK exceptions that are actually interesting are only traceable with "All Exceptions" on.

It's hard for me to comment without a concrete example, but I'm not suggesting you don't use "All Exceptions" at all. My suggestion would be to generally use just "Uncaught Exceptions", and when you are specifically trying to debug an issue that might involve caught exceptions, temporarily turn on "All Exceptions". It's always going to be noisy having this enabled because catching exceptions is not common.

**Comment by ricardo-kowalski** [comment]
The same issue here.

**Comment by jayklaus** [comment]
I had the same problem. Downgrading to version [3.7.11] fixed the issue.
`Flutter downgrade 3.7.11`

**Comment by SandorSzavost28** [comment]
Same Issue after upgrading Flutter 3.10.0

I have 2 different behaviors in my app, 
1. if I "continue" (F5) the http get request is done without errors
2. Error request does not have "response", even if request is to the same address! (2 images attached below of this one)

request including appCheckToken (in my case)
![imagen](https://github.com/flutter/flutter/assets/37272894/24990262-1f1d-4c5c-827e-1d29d6c55af5)

error not getting response
![imagen](https://github.com/flutter/flutter/assets/37272894/35a0a568-a3a4-4744-9291-6913539f53ad)

I hope this help, If you want more details please ask, I'll be happy to contribute

**Comment by DanTup** [comment]
The behaviour from the original post (and many of the recent comments) was explained above:

https://github.com/flutter/flutter/issues/126650#issuecomment-1547909038

This appears to be working as intended. If you have ticked "All Exceptions" in the debug side bar, you are choosing to have the debugger break on all exceptions that are thrown, even if they will be caught be a `catch` block (or `catchError`, etc.). In this case, when the debugger breaks, you need to look at the call stack (or step through as the exception propagates) and understand whether the exception is something that is a problem, or being handled as expected.

I would recommend using "Uncaught exceptions" and not "All exceptions" unless you are specifically trying to debug something involving a caught exception. If you feel that you need to use "All exceptions" to debug common issues, then it's worth filing an issue with details to see whether it's a bug or something that should be caught by "Uncaught exceptions". In some cases an exception is _technically_ caught by the framework, but is treated as _uncaught_ for debugging purposes to make debugging easier. It's possible there are places that this isn't being done, but should.

I have opened https://github.com/dart-lang/sdk/issues/52423 to see whether the specific code here could be amended to not `throw`/`catch` in this case.

If you see are seeing _other_ issues, please file new issues and include clear steps/code to reproduce and a detailed description of the behaviour you're seeing, and the behaviour you expect. There are few other issues mentioned above but without enough information to reproduce, and it's difficult to track multiple different problems in a single flat issue like this.

**Comment by paolanoiaapsystem** [comment]
The same issue here.

**Comment by lionnner** [comment]
The same issue on 3.10.1

**Comment by milindgoel15** [comment]
Well, as long as the release notes do not mention this issue is fixed, it's redundant to say "issue still exists in 3.10.X".  Similar thing for "same issue"

This just spams the issue thread and causes unnecessary notifications to already subscribed participants.

**Comment by nmunoz-mybudgeat** [comment]
I'm experiencing same issue on 3.10.1, would appreciate a fix as this does make debugging difficult when need to break on all exceptions

**Comment by DanTup** [comment]
> this does make debugging difficult when need to break on all exceptions

For those that feel they need to have "all exceptions" always enabled, I'm interested to hear more about the reasons. What sort of exceptions do you need to often debugging that are considered "caught"? I noted [above](https://github.com/flutter/flutter/issues/126650#issuecomment-1547909038) that there are places where exceptions are _technically_ caught (eg. by Flutter to produce detailed error messages) but the VM treats as "uncaught" to improve of debugging. It may be that there are more places it would make sense to do this.

**Comment by cypherchabon** [comment]
The same issue :(

**Comment by Hassanmrwt** [comment]
I am facing the same issue, and have explained in the video 👇👇👇👇
https://youtu.be/lRtrcG_Sz8U

**Comment by Trennum** [comment]
Same issue on 3.10.1. The video that @Hassanmrwt posted is great and demonstrates one of several reasons why a fix for this is necessary and will be very much appreciated.

**Comment by darshankawar** [comment]
Everyone in this thread facing the originally reported issue, please take a look at below comments for your reference:

https://github.com/flutter/flutter/issues/126650#issuecomment-1547909038
https://github.com/flutter/flutter/issues/126650#issuecomment-1551004871

Also, please don't comment `same issue`, but instead upvote original description. 

> If you see are seeing _other_ issues, please file new issues and include clear steps/code to reproduce and a detailed description of the behaviour you're seeing, and the behaviour you expect. There are few other issues mentioned above but without enough information to reproduce, and it's difficult to track multiple different problems in a single flat issue like this.

Please follow this comment and file a new issue if you are seeing behavior other than what is reported originally.

**Comment by brianquinlan** [comment]
This is fixed in https://github.com/dart-lang/sdk/commit/14cb98e874b819428a7a1e8e6702ca46d1b69e4d

I will go through the cherrypick process to see if this can be in Dart 3.0.1.

**Comment by K1yoshiSho** [comment]
@brianquinlan Thanks)

**Comment by brianquinlan** [comment]
@bvoq Your issue is a different one - this issue is related to address parsing, your issue is related to a failure to resolve a host name.

If you think that you've found a new bug, please create a new issue with a complete stack trace.

Thanks!

**Comment by K1yoshiSho** [comment]
@brianquinlan I install latest flutter version 3.10.2 with Dart 3.0.2 and this error not fixed(((

**Comment by milindgoel15** [comment]
Because it's not included yet. Always check the release notes 

https://github.com/flutter/flutter/wiki/Hotfixes-to-the-Stable-Channel#3102-may-24-2023

**Comment by Hassanmrwt** [comment]
I have updated Flutter to the latest available version and it's still there.

**Comment by KristijanZic** [comment]
Can we open this issue again? I'm stuck on the same bug.

**Comment by DanTup** [comment]
This issue has been resolved in the master branch which is why the issue is closed. All issues here are closed when the fix is merged to master (and not when there is a stable release that contains them).

There's an open issue discussing whether to cherry-pick this to stable at https://github.com/dart-lang/sdk/issues/52487.

To re-iterate what's been said above, the behaviour you're seeing is not a bug. If you are ticking "Break on all exceptions" then your debugger will pause on exceptions that are caught/handled. The code has been changed to reduce the noise here, but you can easily work around this issue by using "Break on uncaught exceptions" as standard (which is the default and recommend way), and using "Break on all exceptions" only when trying to debug an exception that is being caught.

If you feel that "Break on uncaught exceptions" is not working well for you and that you need to use "Break on all exceptions" to debug certain types of errors, please detail those issues because it might be possible to improve them. The tooling will always assume you are using "Break on uncaught exceptions" by default, because "Break on all exceptions" will always have "false positives" (by design).

**Comment by K1yoshiSho** [comment]
@brianquinlan I installed latest flutter version 3.10.3 with Dart 3.0.3 and this error not fixed(((
But in changelog: Fixes an error during debugging when InternetAddress.tryParse is used.
https://github.com/flutter/flutter/wiki/Hotfixes-to-the-Stable-Channel#3102-may-24-2023

**Comment by K1yoshiSho** [comment]
![image](https://github.com/flutter/flutter/assets/96978512/2127d698-e3c8-4953-bfac-573d883dbdf8)

**Comment by milindgoel15** [comment]
It looks like a different commit that is merged.

**Comment by darshankawar** [comment]
> There's an open issue discussing whether to cherry-pick this to stable at [dart-lang/sdk#52487](https://github.com/dart-lang/sdk/issues/52487).

This is still open and has not been merged yet. Please follow-up in the linked issue for further updates.

**Comment by DanTup** [comment]
I believe that issue was cherry-picked and included in 3.0.3 (see https://github.com/dart-lang/sdk/commits/stable), however the Cherry-Pick issue was not closed, and there seems to be a typo in the release notes linking to the wrong issue.

However, it seems that the error above is different. It seems like one caught exception was fixed, but there's another very similar. I've added notes to https://github.com/dart-lang/sdk/issues/52487#issuecomment-1576498149 and filed https://github.com/dart-lang/sdk/issues/52615.

That said, I would like to re-iterate again that the default debug settings are "Break on uncaught exceptions" and it will always be assumed this is the setting most people use. If anyone feels they need to use "Break on all exceptions" because the default isn't breaking on some kinds of exceptions they would like it to, _please file issues_. It's unlikely that every instance of caught exceptions will be eliminated, so it would be better to improve "Break on uncaught exceptions" to break where it isn't currently (but you think it should).
